lib/Pulse.Lib.AVLTree.fst

```diff
@@ -0,0 +1,948 @@
+//Pulse AVL tree implementation. Inspired from Steel. The FStar spec file is adopted from Steel
+//----------------------------------------------------------------------------------------------------------
+module Pulse.Lib.AVLTree
+open Pulse.Lib.Pervasives
+module S = Pulse.Lib.Stick.Util
+module FA = Pulse.Lib.Forall.Util
+open FStar.List.Tot
+
+module T = Spec.AVLTree
+
+//type vprop = slprop
+
+let rec is_tree #t ct ft = match ft with
+  | T.Leaf -> pure (ct == None)
+  | T.Node data left' right' ->
+    exists* (p:node_ptr t) (lct:tree_t t) (rct:tree_t t).
+      pure (ct == Some p) **
+      pts_to p { data = data ; left = lct ; right = rct} **
+      is_tree lct left' **
+      is_tree rct right'
+
+//boilerplate$
```

Comment:
There are many such $ ending comments in the file, can we delete them?

Reply:
comment addressed.